app/javascript/lib/imperial.js

```diff
@@ -255,29 +270,47 @@ export default class Imperial {
   }
 
   initialize(action) {
-    let setup;
-    let auctionSetup;
+    this.baseGame = action.payload.baseGame || ImperialEuropeGame.classId;
+    switch (this.baseGame) {
+      case ImperialEuropeGame.classId:
+        this.game = new ImperialEuropeGame();
+        break;
+
+      case Imperial2030Game.classId:
+        this.game = new Imperial2030Game();
+        break;
+
+      case ImperialAsiaGame.classId:
+        this.game = new ImperialAsiaGame();
+        break;
+
+      default:
+        console.error("Undefined gamemode ", this.baseGame);
```

Comment:
Tempted to `throw new Error` instead, but not sure how far down it would break things

Console.error is the next best thing